pennylane/templates/layers/cv_neural_net.py

```diff
@@ -191,28 +154,23 @@ def CVNeuralNetLayers(
         ValueError: if inputs do not have the correct format
     """
 
-    #############
-    # Input checks
-
     wires = Wires(wires)
     repeat = _preprocess(
         theta_1, phi_1, varphi_1, r, phi_r, theta_2, phi_2, varphi_2, a, phi_a, k, wires
     )
 
-    ###############
-
     for l in range(repeat):
-        cv_neural_net_layer(
-            theta_1=theta_1[l],
-            phi_1=phi_1[l],
-            varphi_1=varphi_1[l],
-            r=r[l],
-            phi_r=phi_r[l],
-            theta_2=theta_2[l],
-            phi_2=phi_2[l],
-            varphi_2=varphi_2[l],
-            a=a[l],
-            phi_a=phi_a[l],
-            k=k[l],
-            wires=wires,
-        )
+
+        Interferometer(theta=theta_1[l], phi=phi_1[l], varphi=varphi_1[l], wires=wires)
+
+        r_and_phi_r = [qml.math.expand_dims(r[l], axis=1), qml.math.expand_dims(phi_r[l], axis=1)]
+        r_and_phi_r = qml.math.concatenate(r_and_phi_r, axis=1)
```

Comment:
I think this could be combined into the following?
```python
qml.math.stack([r[l], phi_r[l]], axis=1)
```

E.g.,

```python
>>> a
array([0, 1, 2, 3, 4])
>>> b
array([5, 6, 7, 8])
>>> b = np.array([5, 6, 7, 8, 9])
>>> list(zip(a, b))
[(0, 5), (1, 6), (2, 7), (3, 8), (4, 9)]
>>> qml.math.stack([a, b], axis=1)
array([[0, 5],
       [1, 6],
       [2, 7],
       [3, 8],
       [4, 9]])
```

(I haven't checked that it also works for Torch/TF, but it should unless there is a bug)

Reply:
This way we can reduce 3 function calls down to 1

Reply:
Can verify it works for TF and Torch

Reply:
Oh great! Let me use that. I am not very proficient in tensor processing yet...